keyvalue/record.go

```diff
@@ -112,34 +112,34 @@ func (b *BaseFileRecord) Sys() interface{} {
 }
 
 type runOnceFileRecord struct {
+	modTime time.Time
+
 	record FileRecord
+	data   blob.Blob
+	sys    interface{}
 
-	data     blob.Blob
-	dataErr  error
-	dataDone int64
-	dataOnce sync.Once
+	dataErr     error
+	dirNamesErr error
 
-	dirNames     []string
-	dirNamesErr  error
-	dirNamesOnce sync.Once
+	dirNames []string
 
 	size     int64
-	sizeOnce sync.Once
-
-	mode     hackpadfs.FileMode
-	modeOnce sync.Once
+	dataDone atomic.Int64
```

Comment:
If I remember right, dataDone behaves more like a `bool` but the atomic package didn't have those operations available at the time. Let's try to use `atomic.Bool` here.

Reply:
worth it, sure.